tests/virt_autotest/libvirt_routed_virtual_network.pm

```diff
@@ -70,17 +71,23 @@ sub run_test {
         #attach-interface via virsh worked for all sles guest
         $mac1   = '00:16:3e:32:' . (int(rand(89)) + 10) . ':' . (int(rand(89)) + 10);
         $model1 = (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen')) ? 'netfront' : 'virtio';
-        assert_script_run("virsh attach-interface $guest network vnet_routed --model $model1 --mac $mac1 --live");
+        my $affecter = (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen') && is_sle('=11-sp4')) ? '--persistent' : '';
+        assert_script_run("virsh attach-interface $guest network vnet_routed --model $model1 --mac $mac1 --live $affecter", 60);
 
         $mac2   = '00:16:3e:32:' . (int(rand(89)) + 10) . ':' . (int(rand(89)) + 10);
         $model2 = (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen')) ? 'netfront' : 'virtio';
-        assert_script_run("virsh attach-interface $guest.clone network vnet_routed_clone --model $model2 --mac $mac2 --live");
-
-        test_network_interface("$guest",       mac => $mac1, gate => $gate1, target => $target1, net => "vnet_routed");
-        test_network_interface("$guest.clone", mac => $mac2, gate => $gate2, target => $target2, net => "vnet_routed_clone");
-
-        assert_script_run("virsh detach-interface $guest network --mac $mac1 --current");
-        assert_script_run("virsh detach-interface $guest.clone network --mac $mac2 --current");
+        assert_script_run("virsh attach-interface $guest.clone network vnet_routed_clone --model $model2 --mac $mac2 --live $affecter", 60);
+
+        #Wait for guests attached interface from virtual routed network
+        sleep 30;
```

Comment:
This is not needed and we should avoid `sleep` in general.

Reply:
Good point. I agree that. 
But, I had confirm that after attached virtual interface form **routed network test** to guest system on sles kvm (sles15sp2) host, and then to trigger test_network_interface() immediately, found out some TIMEOUT issue to connect guest via ssh used with ip from routed network. 
please refer to the following link for more details
https://openqa.nue.suse.com/tests/4263937#step/libvirt_routed_virtual_network/105
So, we'd better to add this sleep operation after attached rtual interface form **routed network test** to guest system, and then trigger trigger test_network_interface().